servers/rendering/shader_language.cpp

```diff
@@ -4309,8 +4309,18 @@ ShaderLanguage::DataType ShaderLanguage::get_scalar_type(DataType p_type) {
 		TYPE_INT,
 		TYPE_UINT,
 		TYPE_FLOAT,
+		TYPE_INT,
+		TYPE_UINT,
+		TYPE_FLOAT,
+		TYPE_INT,
+		TYPE_UINT,
+		TYPE_FLOAT,
+		TYPE_FLOAT,
```

Comment:
I would simply replace all of them with TYPE_VOID, since they cannot be represented by a scalar value.

Reply:
I was trying to match the original commit, which assigned TYPE_SAMPLER2D -> TYPE_FLOAT, TYPE_ISAMPLER2D -> TYPE_INT, and TYPE_USAMPLER2D -> TYPE_UINT. I'm not sure, but I don't think this is required anywhere, so I guess yes all samplers can be set to TYPE_VOID.